src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseGradingResource.java

```diff
@@ -94,4 +97,28 @@ public ProgrammingExerciseGradingResource(ProgrammingExerciseGradingService prog
         resultRepository.saveAll(updatedResults);
         return ResponseEntity.ok(updatedResults.size());
     }
+
+    /**
+     * Get the exercise's test case statistics for the the given exercise id.
+     *
+     * @param exerciseId of the the exercise.
+     * @return the test case statistics for the exercise.
+     */
+    @GetMapping(STATISTICS)
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ProgrammingExerciseGradingStatisticsDTO> getGradingStatistics(@PathVariable Long exerciseId) {
+        log.debug("REST request to get test case statistics for programming exercise {}", exerciseId);
+        ProgrammingExercise programmingExercise = programmingExerciseService.findWithTemplateParticipationAndSolutionParticipationById(exerciseId);
+
+        Course course = programmingExercise.getCourseViaExerciseGroupOrCourseMember();
+        User user = userService.getUserWithGroupsAndAuthorities();
+
+        if (!authCheckService.isAtLeastTeachingAssistantInCourse(course, user)) {
```

Comment:
I guess this should be "at lest instructor" based on the `@PreAuthorize`

Reply:
This is fixed [4a1c028](https://github.com/ls1intum/Artemis/pull/2172/commits/4a1c028de80261f3ae08b605aab58e458489b1e6)